### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by hixi**
Make use of docker to setup and run this project (using docker-compose).

I'm open to keep this change on my repository only, as it doesn't really add any features to cf_audit; in that case, just close this PR.

### CODE REVIEW

File: app.wsgi [hunk 1]
```diff
@@ -0,0 +1 @@
+from www import app as application
```
Comment (Zverik) [inline app.wsgi]: We've got `cf_audit.wsgi` already.

File: config.py [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
 BASE_DIR = os.path.abspath(os.path.dirname(__file__))
 
-DEBUG = True
+DEBUG = os.environ.get('DEBUG', True)
 
-DATABASE_URI = 'sqlite:///' + os.path.join(BASE_DIR, 'audit.db')
+DATABASE_URI = os.environ.get('DATABASE_URI', 'sqlite:///' + os.path.join(BASE_DIR, 'audit.db'))
 # DATABASE_URI = 'postgresql://localhost/cf_audit'
-MAX_CONTENT_LENGTH = 16*1024*1024
+MAX_CONTENT_LENGTH = int(os.environ.get('MAX_CONTENT_LENGTH', 16*2014*1024))
```
Comment (Zverik) [inline config.py]: 2014?

File: config.py [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
 
-ADMINS = set([290271])  # Zverik
+ADMINS = set(os.environ.get('ADMINS', '').split(','))
+if not ADMINS:
```
Comment (Zverik) [inline config.py]: I think you can remove this, otherwise people would get myself as an admin. Maybe throw an exception?

File: config.py [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
 
-# Override these (and anything else) in config_local.py
-OAUTH_KEY = ''
-OAUTH_SECRET = ''
-SECRET_KEY = 'sdkjfhsfljhsadf'
+# Override these (and anything else) in config_local.py or 
+# set environment variable accordingly.
+OAUTH_KEY = os.environ.get('OAUTH_KEY', '')
+OAUTH_SECRET = os.environ.get('OAUTH_SECRET', '')
+SECRET_KEY = os.environ.get('SECRET_KEY', 'sdkjfhsfljhsadf')
```
Comment (Zverik) [inline config.py]: Again, having a default value is unsafe here.

File: config.py [hunk 5]
```diff
@@ -1,18 +1,21 @@
 import os
 BASE_DIR = os.path.abspath(os.path.dirname(__file__))
 
-DEBUG = True
+DEBUG = os.environ.get('DEBUG', True)
 
-DATABASE_URI = 'sqlite:///' + os.path.join(BASE_DIR, 'audit.db')
+DATABASE_URI = os.environ.get('DATABASE_URI', 'sqlite:///' + os.path.join(BASE_DIR, 'audit.db'))
 # DATABASE_URI = 'postgresql://localhost/cf_audit'
```
Comment (Zverik) [inline config.py]: Since there is an example in the docker file, you can remove this line.

File: migrate.py [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env python3
```
Comment (Zverik) [inline migrate.py]: Note that cf_audit is tested on python 2. It may need porting to python3.

File: app.wsgi (hunk 1 above)
Comment (hixi) [reply app.wsgi]: OK, let's use that one, even though adding `BASE_DIR` isn't necessary in this case.

File: config.py (hunk 2 above)
Comment (hixi) [reply config.py]: Might perhaps been a good year for wine, but is completely wrong here.

File: migrate.py (hunk 6 above)
Comment (hixi) [reply migrate.py]: Will move to Python 2 with the entire Dockerfile. I was running this on Python 3 the entire time, and it worked so far, so porting shouldn't be a too big a hassle at a later point (Python 2 will be retired in 2020, IIRC).

By tested do you mean with automated tests or manually by running it? I couldn't find any tests, if there are, I should probably start running them.

File: config.py (hunk 5 above)
Comment (hixi) [reply config.py]: :+1:

File: config.py (hunk 3 above)
Comment (hixi) [reply config.py]: I'll see what I can do and submit a changed version, best combined with the `OAUTH_KEY` and similar settings. Might need further refinement.

File: config.py (hunk 4 above)
Comment (hixi) [reply config.py]: It is, I just tried mimicking the behaviour it had before. I'll gladly remove this.

File: migrate.py (hunk 6 above)
Comment (Zverik) [reply migrate.py]: No, I mean on my server. It'd be good to have it running on python 3, though.

Comment (hixi) [reply migrate.py]: I'll check if I can find a mean to validate with 2to3 or something similar, for the time being, let's stick to Python 2 (even though it ran OK using Python 3 I'm not convinced there were no issues at all). If I get to the point where I can be pretty sure all is well I'll make another pull request, if that is OK for you.

**Pull request merged**